src/vs/platform/state/electron-main/stateMainService.ts

```diff
@@ -151,12 +151,12 @@ export class StateMainService implements IStateMainService {
 
 	private static readonly STATE_FILE = 'storage.json';
 
-	private readonly stateFilePath = joinPath(this.environmentMainService.globalStorageHome, StateMainService.STATE_FILE);
+	private readonly stateFilePath = joinPath(this.userDataProfilesService.defaultProfile.globalStorageHome, StateMainService.STATE_FILE);
 
 	private readonly fileStorage: FileStorage;
 
 	constructor(
-		@IEnvironmentMainService private readonly environmentMainService: IEnvironmentMainService,
+		@IUserDataProfilesService private readonly userDataProfilesService: IUserDataProfilesService,
```

Comment:
State service shall not depend on profile service. We shall have an stateStorateHome on environment service.